entity_version.post_update.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @file
+ * Entity version post updates.
+ */
+
+declare(strict_types = 1);
+
+/**
+ * Apply entity version settings for bundles with version field.
+ */
+function entity_version_post_update_configure_settings() {
+  // Introduced new configuration to select a version field of an entity bundle
+  // as the main version field to use by the (sub)module functionalities,
+  // instead of searching for all version fields on the bundle and using the
+  // first one.
+  $versioned_entity_types = \Drupal::service('entity_field.manager')->getFieldMapByFieldType('entity_version');
+  // Get entity types and bundles where the entity_version field is present.
+  $version_settings_storage = \Drupal::entityTypeManager()->getStorage('entity_version_settings');
+
+  // Loop through the mapping and create config entities if they don't exist.
+  foreach ($versioned_entity_types as $entity_type_id => $fields) {
+    foreach ($fields as $field_name => $bundle_info) {
+      foreach ($bundle_info['bundles'] as $bundle_name) {
```

Comment:
This is also prone to errors. A field could be present in two bundles, but one of the two bundles has two fields. We don't ensure that the correct one is picked.